src/traces/heatmap/clean_2d_array.js

```diff
@@ -10,7 +10,7 @@
 
 var isNumeric = require('fast-isnumeric');
 
-module.exports = function clean2dArray(zOld, transpose) {
+module.exports = function clean2dArray(zOld, transpose, trace, xa, ya) {
```

Comment:
Minor style tip: no need here to pass `transpose` and `trace` as `transpose = trace.transpose`. 

One example of a better call signature:

```js
function clean2dArray(zOld, trace, xa, ya) { /* */ }

Reply:
Done in 59edb5704f076b438e6b8460b3e03097d61748ab